torchbenchmark/util/model.py

```diff
@@ -91,6 +91,20 @@ def __init__(self, test: str, device: str, jit: bool=False, batch_size: Optional
         self.run_contexts = [
             enable_profiling_executor  # force JIT profiling executor to be enabled by default
         ]
+
+        # taken from torchdynamo benchmarks, this further controls randomness settings
+        def deterministic_torch_manual_seed(*args, **kwargs):
+            from torch._C import default_generator
+
+            seed = 1337
+            import torch.cuda
+
+            if not torch.cuda._is_in_bad_fork():
```

Comment:
what's this- any idea if you're in it?

Reply:
for my test i used 

```
def reset_rng_state():
    torch.manual_seed(1337)
    random.seed(1337)
    np.random.seed(1337)
```

any idea if you're using np or random libs in model init and therefore missing those?

Reply:
one debugging technique could be to add a function that pulls a random  number from torch and prints it; then call this function from various points in your benchmark (same points in both ref and test).  you could start at coarse grain, e.g. put these  calls at various places in the torchbench-distributed-ddp-experiment code, before/after calling the model etc.  then you could even experiment with adding it as a hook after each layer in the model or something like that.

Reply:
hmm, I should take a look at this - it could be the problem, seems like it has something to do with forking processes. https://github.com/pytorch/pytorch/blob/master/torch/cuda/__init__.py#L228-L230.

re: reset_rng_state(), I think this is the same thing that set_random_seed() does: https://github.com/pytorch/benchmark/blob/main/torchbenchmark/util/env_check.py#L16-L22